src/NUnitFramework/framework/Internal/TestExecutionContext.cs

```diff
@@ -360,6 +365,19 @@ internal IWorkItemDispatcher Dispatcher
         public ParallelScope ParallelScope { get; set; }
 
         /// <summary>
+        /// The unique name of the worker that spawned the context.
+        /// For builds with out the parallel feature, it is null.
+        /// </summary>
+        public string WorkerId {
+            get {
+                if (_workerId != null) return _workerId;
+                if (_priorContext != null) return _priorContext.WorkerId;
+                return null;
+            }
+            internal set {_workerId = value;}
+        }
```

Comment:
This surprised me. I figured you would set the WorkerId in CompositeWorkItem.CreateChildWorkItems. This is good but to reduce confusion, please add a comment explaining that some items are executed directly by their parents under the same agent.